### ISSUE TITLE
Developer Documentation

### ISSUE THREAD

**Issue by posixeleni** [IQSS/dataverse#912]
@scolapasta I need your help to make sure that these are the main dev areas that we need to make sure are well documented for 4.0.

* Build Environment (Configuring NetBeans)
   * Build for the 1st time
* Application Environment (Configuring GlassFish and PostgreSQL)
* Developing / Contributing with Git

**Comment by scolapasta** [comment]
Yes, those do seem to be the main issues. Assigning back to @posixeleni unless you need something else from me.

**Comment by pdurbin** [comment]
I'd love to see mention of our emerging coding style/standards per #775.

**Comment by posixeleni** [comment]
@scolapasta I cant write the developer documentation so I'm not sure why this is assigned to me? Is there a way we can break these sections up by developers (assign tickets to individual devs for each section that documentation is needed) that are working on them so they can complete these. I am out for the next week so wont be able to look at this till then.

**Comment by pdurbin** [comment]
I spent probably 10 hours on http://guides.dataverse.org/en/latest/developers and I'm happy enough with the content and how it looks. Good enough to tag 4.0 I think.

In particular, I attempted to address the desires of people who responded to this thread I started on the community list: [What would you like to see in the Dataverse Developers Guide?](https://groups.google.com/forum/#!topic/dataverse-community/K4mO9NJTxa0)

Passing to @scolapasta rather than QA since he's in the best position to close this issue.

@scolapasta please check that the **Branching Strategy** section makes sense, especially.

**Comment by pdurbin** [comment]
@abidart as we discussed in our weekly meeting, please give us feedback on http://guides.dataverse.org/en/latest/developers

Here's the source, by the way: https://github.com/IQSS/dataverse/tree/master/doc/sphinx-guides/source/developers

**Comment by pdurbin** [comment]
@abidart did a great job ensuring the dev guide explains enough for a new developer to get Dataverse running locally. We learned that the installer even works on Ubuntu! At least for dev (#1059 is for prod).

I just added a few finishing touches and did a build so http://guides.dataverse.org/en/latest/developers has the latest.

@kcondon if you do have time to look through the dev guide I'd appreciate it! Passing to QA.
